app/routes/questions.$questionId.$.tsx

```diff
@@ -96,6 +96,13 @@ export default function RenderArticle() {
     setShowNav(false)
   }, [location.key])
 
+  useEffect(() => {
+    data.then((val) => {
+      const question = val as Question
+      if (question.title) document.title = question.title
+    })
```

Comment:
I think ideally this `as Question` cast would be removed as it breaks type safety and makes it more likely that a bug would be introduced if the return type of the loader is changed. This seems better IMHO:
```suggestion
    data.then((val) => {
      if ('title' in val) document.title = val.title
    })
```